lib/packwerk/checker.rb

```diff
@@ -22,11 +22,29 @@ def included(base)
       def all
         T.unsafe(@checkers).map(&:new)
       end
+
+      sig { params(violation_type: String).returns(Checker) }
+      def find(violation_type)
+        checker_by_violation_type(violation_type)
+      end
+
+      private
+
+      sig { params(name: String).returns(Checker) }
+      def checker_by_violation_type(name)
+        @checker_by_violation_type ||= T.let(Checker.all.map do |c|
```

Comment:
```suggestion
        @checker_by_violation_type ||= T.let(Checker.all.to_h do |checker|
```
You can use Array#to_h to map and cast in one call.